drivers/hmis/app/models/hmis/hud/validators/enrollment_validator.rb

```diff
@@ -0,0 +1,47 @@
+class Hmis::Hud::Validators::EnrollmentValidator < Hmis::Hud::Validators::BaseValidator
+  IGNORED = [
+    :ExportID,
+    :DateCreated,
+    :DateUpdated,
+    :ProjectID, # allowed to be null if wip record is present
+  ].freeze
+
+  def configuration
+    Hmis::Hud::Enrollment.hmis_configuration(version: '2022').except(*IGNORED)
+  end
+
+  def self.validate_entry_date(entry_date, enrollment:, options: {})
+    return [] unless entry_date.present?
+
+    errors = HmisErrors::Errors.new
+    dob = enrollment.client&.dob
+    exit_date = enrollment&.exit_date
+
+    errors.add :entry_date, :out_of_range, message: future_message, **options if entry_date.future?
+    errors.add :entry_date, :out_of_range, message: over_twenty_years_ago_message, **options if entry_date < (Date.today - 20.years)
+    errors.add :entry_date, :out_of_range, message: before_dob_message, **options if dob.present? && dob > entry_date
+    errors.add :entry_date, :out_of_range, message: after_exit_message(exit_date), **options if exit_date.present? && exit_date < entry_date
+
+    unless enrollment.head_of_household?
+      hoh_entry_date = enrollment.hoh_entry_date
+      errors.add :entry_date, :out_of_range, severity: :warning, message: before_hoh_entry_message(hoh_entry_date), **options if hoh_entry_date.present? && entry_date < hoh_entry_date
+    end
```

Comment:
This warns "entry date cannot be before HoH entry date."

It should probably be an error IF we are validating in the context of (1) a new household member being added (ie a new WIP enrollment), or (2) a WIP intake assessment being saved or submitted for the first time.

Reply:
Maybe, though isn't it a problem whenever it happens?

Reply:
Is it? If the HoH changed, then the old HoH's entry date would be before the HoH's entry date. If you are going back and doing data correction in that persons intake assessment, you shouldn't get an error (I think).
This is still assuming that we are _not_ doing entry date back-dating when the HoH changes.

Reply:
I'm probably not following completely, but current guidance would be that the HoH Entry Date must be on or before any other members.  
As you mentioned, if we assume we aren't back-dating, then it should _not_ be an error, but the guidance is still that the HoH must be there for the full enrollment time.

Reply:
Right. That's why I made it be a _warning_, not an error. I was just pointing out that in certain cases, we probably do want to treat it as an error. It is something I am planning to come back to for 184407413, I hadn't nailed down the logic well enough to implement it here

Reply:
Sounds good, and sorry, I was using error and warning interchangeably.